lib/pdk/cli/exec/command.rb

```diff
@@ -0,0 +1,233 @@
+
+              raise PDK::CLI::FatalError, _('Current working directory is not part of a module. (No metadata.json was found.)')
+            end
+
+            if Dir.pwd == mod_root || context == :pwd
+              run_process_in_clean_env!
+            else
+              Dir.chdir(mod_root) do
+                run_process_in_clean_env!
+              end
+            end
+          else
+            run_process!
+          end
+
+          # Stop spinning when done (if configured).
+          stop_spinner
+
+          @stdout.rewind
+          @stderr.rewind
+
+          process_data = {
+            stdout: @stdout.read,
+            stderr: @stderr.read,
+            exit_code: @process.exit_code,
+            duration: @duration,
+          }
+
+          PDK.logger.debug _('STDOUT: %{output}') % {
+            output: process_data[:stdout].empty? ? 'N/A' : "\n#{process_data[:stdout]}",
+          }
+          PDK.logger.debug _('STDERR: %{output}') % {
+            output: process_data[:stderr].empty? ? 'N/A' : "\n#{process_data[:stderr]}",
+          }
+
+          process_data
+        ensure
+          @stdout.close
+          @stderr.close
+        end
+
+        protected
+
+        def warn_on_legacy_env_vars!
+          if ENV['PUPPET_GEM_VERSION']
+            PDK.logger.warn_once _(
+              'PUPPET_GEM_VERSION is not supported by PDK. ' \
+              'Use the --puppet-version option on your PDK command ' \
+              'or set the PDK_PUPPET_VERSION environment variable instead',
+            )
+          end
+
+          %w[FACTER HIERA].each do |gem|
+            if ENV["#{gem}_GEM_VERSION"]
+              PDK.logger.warn_once _('%{varname} is not supported by PDK.') % { varname: "#{gem}_GEM_VERSION" }
+            end
+          end
+        end
+
+        def resolved_env_for_command
+          warn_on_legacy_env_vars!
+
+          resolved_env = {}
+
+          resolved_env['TERM'] = ENV['TERM']
+          resolved_env['PUPPET_GEM_VERSION'] = nil
+          resolved_env['FACTER_GEM_VERSION'] = nil
+          resolved_env['HIERA_GEM_VERSION'] = nil
+
+          resolved_env.merge!(@environment.dup)
+
+          resolved_env['BUNDLE_IGNORE_CONFIG'] = '1'
+
+          if [:module, :pwd].include?(context)
+            resolved_env['GEM_HOME'] = PDK::Util::RubyVersion.gem_home
+            resolved_env['GEM_PATH'] = PDK::Util::RubyVersion.gem_path
+
+            # Make sure invocation of Ruby prefers our private installation.
+            package_binpath = PDK::Util.package_install? ? File.join(PDK::Util.pdk_package_basedir, 'bin') : nil
+
+            resolved_env['PATH'] = [
+              PDK::Util::RubyVersion.bin_path,
+              File.join(resolved_env['GEM_HOME'], 'bin'),
+              PDK::Util::RubyVersion.gem_paths_raw.map { |gem_path| File.join(gem_path, 'bin') },
+              package_binpath,
+              PDK::Util.package_install? ? PDK::Util::Git.git_paths : nil,
+              ENV['PATH'],
```

Comment:
Warning: ENV has known issues with non-ASCII characters on Windows environment variables.  https://github.com/puppetlabs/puppet/blob/eb19bedb221959bd202e598401a3d29ea77950c9/spec/unit/util_spec.rb#L99-L218

Typically this comes from non-English Usernames e.g. `C:\Users\jmüller`

Reply:
Similar to above, good thing to know but shouldn't be a regression since this is the existing implementation.